src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -374,9 +374,10 @@ internal override bool HasPointerType
                 if (_lazyType != null)
                 {
                     Debug.Assert(_lazyType.Value.DefaultType.IsPointerType() ==
-                        IsPointerFieldSyntactically());
+                        IsPointerFieldSyntactically() ||
+                        _lazyType.Value.DefaultType.Kind == SymbolKind.FunctionPointer);
 
-                    return _lazyType.Value.DefaultType.IsPointerType();
+                    return _lazyType.Value.DefaultType.IsPointerType() || _lazyType.Value.DefaultType.Kind == SymbolKind.FunctionPointer;
```

Comment:
>_lazyType.Value.DefaultType.IsPointerType() || _lazyType.Value.DefaultType.Kind == SymbolKind.FunctionPointer [](start = 27, length = 109)

Consider using switch expression over symbol kind, also putting the value into the local and using the local in the assert above (what we return here and the value we get from ```IsPointerFieldSyntactically``` should match).